### PR TITLE
do not return control to main program once DFU download starts

### PR DESCRIPTION
**Pull request by alastairpatrick**

### CODE REVIEW

File: src/ice_usb.c [hunk 1]
```diff
@@ -209,15 +225,21 @@ void tud_dfu_download_cb(uint8_t alt, uint16_t block_num, const uint8_t *data, u
     }
 
     tud_dfu_finish_flashing(DFU_STATUS_OK);
+
+    // Never return to the main loop.
+    if (!started) {
+        started = true;
+        for (;;) {
+            tud_task();
```
Comment (josuah) [inline src/ice_usb.c:233]: Good idea!

File: src/ice_usb.c [hunk 2]
```diff
@@ -36,6 +37,8 @@
 #include "tinyuf2/uf2.h"
 #include "tinyuf2/board_api.h"
 
+#define WATCHDOG_DELAY 3000
```
Comment (josuah) [inline src/ice_usb.c:40]: Is that 3s? Is it not a bit short?
Flash programming takes more time, and a slow host could make the transfer arbitrarily long.

**Review by josuah — APPROVED** [review]
I have a few questions, but nothing preventing merging now. 
I did not know how to handle this, and this approach sounds most reasonable.

File: src/ice_usb.c (hunk 2 above)
Comment (alastairpatrick) [reply src/ice_usb.c:40]: The tud_dfu_download_cb callback invokes watchdog_update(), which resets the watchdog timer each time. So it should only fire if tud_dfu_download_cb is not invoked again within 3 secs, which I think should be okay even for flash.

Comment (josuah) [reply src/ice_usb.c:40]: I missed that somehow, perfect then! 3s of idle between two operations suggests something got stuck, sane to reboot anyway.

**Comment by josuah** [comment]
Merged in from command-line.
Thanks!

**Pull request merged**